lib/java/com/google/android/material/datepicker/MaterialCalendarGridView.java

```diff
@@ -152,7 +153,9 @@ protected final void onDraw(@NonNull Canvas canvas) {
                     : getChildAt(firstHighlightPosition - 1).getLeft();
       } else {
         dayCompute.setTimeInMillis(startItem);
-        firstHighlightPosition = monthAdapter.dayToPosition(dayCompute.get(Calendar.DAY_OF_MONTH));
+        int day = dayCompute.get(Calendar.DAY_OF_MONTH);
+        int dayPosition = monthAdapter.dayToPosition(day);
```

Comment:
This logic being repeated again down below. Consider pulling the logic into a method that will accept the time in milliseconds and return the day position.

Reply:
Hey Raaj, let's not merge the pull request. I have a more comprehensive fix pending for review.

Reply:
Sounds good. thanks for letting me know.